### PR TITLE
Remove redundant bbox slice in plot_predictions

### PR DESCRIPTION
**Pull request by raimbekovm**
Removed redundant [:, :4] slice when converting bboxes to xywh format.

Changes:
- Simplified line 357: batched_preds["bboxes"] = ops.xyxy2xywh(batched_preds["bboxes"])
- Removed TODO comment as issue is now resolved

The batched_preds["bboxes"] tensor always has exactly 4 columns from postprocess() method (line 126), making the [:, :4] slice unnecessary. The ops.xyxy2xywh() function already asserts input has 4 columns, confirming this change is safe.

## Testing Commands

Validated with `plots=True` for detect/seg/pose - all `val_batch*_pred.jpg` outputs generated correctly:

```bash
# Detection
yolo val model=yolo11n.pt data=coco8.yaml plots=True

# Segmentation  
yolo val model=yolo11n-seg.pt data=coco8-seg.yaml plots=True

# Pose Estimation
yolo val model=yolo11n-pose.pt data=coco8-pose.yaml plots=True
```

Results saved to `runs/{detect,segment,pose}/val/` with correct bbox visualizations, mask overlays, and keypoint skeletons respectively.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Simplifies bbox format conversion in `plot_predictions` by removing an unnecessary slice and converting the full tensor directly ✅

### 📊 Key Changes
- Replaced in-place slicing assignment on `batched_preds["bboxes"][:, :4]` with a full-tensor conversion via `ops.xyxy2xywh(batched_preds["bboxes"])`
- Removed a stale `# TODO: fix this` comment tied to the redundant slicing logic

### 🎯 Purpose & Impact
- Reduces redundant indexing and in-place mutation, making the code clearer and less error-prone 🧹
- Ensures consistent bbox conversion behavior for the full `bboxes` tensor, which can improve maintainability and avoid subtle shape/format mismatches 🔧

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @raimbekovm, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). This is an automated message, and an Ultralytics engineer will assist with review shortly 😊🚀

File: ultralytics/models/yolo/detect/val.py [hunk 1]
```diff
@@ -353,8 +353,7 @@ def plot_predictions(
         keys = preds[0].keys()
         max_det = max_det or self.args.max_det
         batched_preds = {k: torch.cat([x[k][:max_det] for x in preds], dim=0) for k in keys}
-        # TODO: fix this
-        batched_preds["bboxes"][:, :4] = ops.xyxy2xywh(batched_preds["bboxes"][:, :4])  # convert to xywh format
+        batched_preds["bboxes"] = ops.xyxy2xywh(batched_preds["bboxes"])  # convert to xywh format
```
Comment (UltralyticsAssistant) [inline ultralytics/models/yolo/detect/val.py:357]: ⚠️ **HIGH**: This changes the contract from “convert only the first 4 bbox coords” to “convert the entire tensor”. The prior `[:, :4]` strongly suggests `batched_preds["bboxes"]` can have extra trailing columns (e.g., scores/cls/track id). Passing the full tensor to `ops.xyxy2xywh()` may either error (if it expects `(...,4)`) or incorrectly transform non-coordinate fields. Consider converting only the first 4 columns while preserving any extras, without in-place mutation.

**Suggested change:**
```suggestion
        batched_preds["bboxes"] = torch.cat((ops.xyxy2xywh(batched_preds["bboxes"][..., :4]), batched_preds["bboxes"][..., 4:]), dim=-1) if batched_preds["bboxes"].shape[-1] > 4 else ops.xyxy2xywh(batched_preds["bboxes"])  # convert to xywh format
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good intent (removing redundant slicing/in-place mutation), but the previous `[:, :4]` strongly implies `bboxes` may contain more than 4 values per row. Converting the full tensor risks breaking shape expectations or corrupting non-coordinate fields; preserve extra columns while converting only the first four coordinates.

💬 Posted 1 inline comment

**Comment by raimbekovm** [comment]
`batched_preds["bboxes"]` always has exactly 4 columns here:

- `DetectionValidator.postprocess()` (line 126) returns `{"bboxes": x[:, :4], ...}`
- `ops.xyxy2xywh()` has `assert x.shape[-1] == 4` - fails if >4 columns
- `OBBValidator` (5 columns) overrides `plot_predictions()` completely

The `[:, :4]` was redundant (note the `# TODO: fix this` comment).

**Comment by glenn-jocher** [comment]
Thanks for the clarification — before we merge, can you double-check that every codepath that calls this `plot_predictions()` always passes `batched_preds["bboxes"]` as shape `(N, 4)` (i.e. no extra columns like track IDs or appended metadata), since [`ops.xyxy2xywh()` asserts the last dim is 4](https://docs.ultralytics.com/reference/utils/ops/#ultralytics.utils.ops.xyxy2xywh) and the prior `[:, :4]` would have silently tolerated wider tensors.

**Comment by raimbekovm** [comment]
Double-checked all codepaths - `batched_preds["bboxes"]` is always **(N, 4)** for validators using this method:

- `DetectionValidator`: `postprocess()` returns `{"bboxes": x[:, :4], ...}` (line 126)
- `SegmentationValidator`: inherits, uses `extra` for masks
- `PoseValidator`: inherits, uses `extra` for keypoints  
- `RTDETRValidator`: own `postprocess()` returns `{"bboxes": x[:, :4], ...}` (line 181)
- `NASValidator`: inherits from Detection

`OBBValidator` has 5 columns (xywhr) but completely overrides `plot_predictions()` (lines 145-172) and doesn't call the parent method.

Tracking IDs are not used in validation pipeline - they're stored in `results.boxes.id`, not appended to bboxes.

The `[:, :4]` was defensive programming from a refactor (hence `# TODO: fix this`). `ops.xyxy2xywh()` assert provides the same safety.

**Comment by glenn-jocher** [comment]
Confirmed, thanks — that resolves my concern; could you also run a quick `yolo val ... plots=True` for detect/seg/pose to ensure `DetectionValidator.plot_predictions()` still saves the expected `val_batch*_pred.jpg` outputs, and then we should be good to merge (see the [DetectionValidator.plot_predictions reference](https://docs.ultralytics.com/reference/models/yolo/detect/val/)).

**Comment by raimbekovm** [comment]
Confirmed - ran validation with `plots=True` for detect/seg/pose:

- **Detection** (`yolo11n.pt`): `val_batch*_pred.jpg` generated with correct bbox visualizations
- **Segmentation** (`yolo11n-seg.pt`): Prediction plots with boxes and mask overlays
- **Pose** (`yolo11n-pose.pt`): Prediction plots with keypoints and skeletons

All outputs saved correctly to `runs/{detect,segment,pose}/val/`. No errors during bbox format conversion.

**Comment by glenn-jocher** [comment]
Thanks for running those checks across detect/seg/pose—this looks good from our side; once CI is green I’ll proceed, and if you can paste the exact `yolo val ... plots=True` commands you used into the PR description it’ll help future readers (see the [Val mode docs](https://docs.ultralytics.com/modes/val/)).

**Comment by raimbekovm** [comment]
Added testing commands to the PR description. Used coco8 mini-datasets for quick validation:

```bash
yolo val model=yolo11n.pt data=coco8.yaml plots=True
yolo val model=yolo11n-seg.pt data=coco8-seg.yaml plots=True
yolo val model=yolo11n-pose.pt data=coco8-pose.yaml plots=True
```

All three generated the expected `val_batch*_pred.jpg` outputs in `runs/{detect,segment,pose}/val/` directories.

**Comment by glenn-jocher** [comment]
Perfect, thanks for adding the exact `yolo val ... plots=True` repro commands to the PR description—once the PR’s CI checks are green we’ll proceed with merging.
